cli/tests/test_cat_command.rs

```diff
@@ -82,3 +98,25 @@ fn test_cat() {
     >>>>>>>
     "###);
 }
+
+#[cfg(unix)]
+#[test]
+fn test_cat_symlink() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "repo"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    std::fs::write(repo_path.join("file1"), "a\n").unwrap();
+    std::fs::create_dir(repo_path.join("dir")).unwrap();
+    std::fs::write(repo_path.join("dir").join("file2"), "c\n").unwrap();
+    std::os::unix::fs::symlink("d", repo_path.join("symlink1")).unwrap();
```

Comment:
I don't see where this "d" file gets created. Where is that done?

Reply:
Maybe I misunderstood and this is just printing the target. A different name like `symlink_target` would make that clearer.

Reply:
Sorry, still thinking about this. It would be nicer if we printed something that made it clear that the target is a symlink. Perhaps something similar to what `ls -al` does? The same format should probably be used for `diff` as well.

```
symlink1@ -> d
```

Could you also please document how symlinks are handled in the `--help` output?

Reply:
I'll update the symlink target.

> It would be nicer if we printed something that made it clear that the target is a symlink. Perhaps something similar to what `ls -al` does?

I don't think `jj cat` should mutate the output content.

Reply:
> I don't think jj cat should mutate the output content.

Sure, but is a symlink literally a file containing the link target as a string? Don't you have to use [`readlink()`](https://man7.org/linux/man-pages/man2/readlink.2.html) to get the target? In that case, isn't the representation of the content somewhat up to us?

Maybe it is better to just print the path as-is, but it's unfortunate that there's no way to tell that the file is a symlink from the output.

Reply:
> Sure, but is a symlink literally a file containing the link target as a string?

I think that's valid in this context. jj (and git) manages link target as data, and `jj diff symlink` shows difference of the target values, not the changes in the target file.

Anyway, the last patch isn't important. I'll split it to a separate PR.